python/plugins/processing/algs/qgis/Rasterize.py

```diff
@@ -0,0 +1,282 @@
+    """Processing algorithm renders map canvas to a raster file.
+    It's possible to choose the following parameters:
+        - Map theme to render
+        - Layer to render
+        - The minimum extent to render
+        - The tile size
+        - Map unit per pixel
+        - The output (can be saved to a file or to a temporary file and
+          automatically opened as layer in qgis)
+    """
+
+    # Constants used to refer to parameters and outputs. They will be
+    # used when calling the algorithm from another algorithm, or when
+    # calling from the QGIS console.
+
+    OUTPUT_LAYER = 'OUTPUT_LAYER'
+    MAP_THEME = 'MAP_THEME'
+    LAYER = 'LAYER'
+    EXTENT = 'EXTENT'
+    TILE_SIZE = 'TILE_SIZE'
+    MAP_UNITS_PER_PIXEL = 'MAP_UNITS_PER_PIXEL'
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        """Here we define the inputs and output of the algorithm, along
+        with some other properties.
+        """
+
+        # The parameters
+        self.addParameter(
+            QgsProcessingParameterString(self.MAP_THEME,
+                                         description=self.tr(
+                                             'Map theme to render.'),
+                                         defaultValue=None, optional=True))
+
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(self.LAYER, description=self.tr(
+                'Layer to render. Will only be used if the map theme is not '
+                'set. '
+                'If both, map theme and layer are not '
+                'set, the current map content will be rendered.'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterExtent(self.EXTENT, description=self.tr(
+                'The minimum extent to render. Will internally be extended to '
+                'be '
+                'a multiple of the tile sizes.')))
+        self.addParameter(
+            QgsProcessingParameterNumber(self.TILE_SIZE, self.tr('Tile size'),
+                                         defaultValue=1024))
+        self.addParameter(QgsProcessingParameterNumber(self.MAP_UNITS_PER_PIXEL,
+                                                       self.tr(
+                                                           'Map units per '
+                                                           'pixel'),
+                                                       defaultValue=100))
+
+        # We add a raster layer as output
+        self.addParameter(QgsProcessingParameterRasterDestination(
+            self.OUTPUT_LAYER,
+            self.tr(
+                'Output layer')))
+
+    def name(self):
+        # Unique (non-user visible) name of algorithm
+        return 'Rasterize'
+
+    def displayName(self):
+        # The name that the user will see in the toolbox
+        return self.tr('Convert map to raster')
+
+    def group(self):
```

Comment:
Can you also implement the tags method please? See DeleteColumn.py for an example